testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -6188,6 +6188,29 @@ void checkRegexpExtract(SqlOperatorFixture f0, FunctionAlias functionAlias) {
     f.checkNull("log(10, cast(null as real))");
   }
 
+  @Test void testLog2Func() {
+    final SqlOperatorFixture f0 = Fixtures.forOperators(true);
+    f0.setFor(SqlLibraryOperators.LOG2, VmName.EXPAND);
+    f0.checkFails("^log2(4)^",
+        "No match found for function signature LOG2\\(<NUMERIC>\\)", false);
+    final SqlOperatorFixture f = f0.withLibrary(SqlLibrary.MYSQL);
+    f.checkScalarApprox("log2(2)", "DOUBLE NOT NULL",
+        isWithin(1.0, 0.000001));
+    f.checkScalarApprox("log2(4)", "DOUBLE NOT NULL",
+        isWithin(2.0, 0.000001));
+    f.checkScalarApprox("log2(65536)", "DOUBLE NOT NULL",
+        isWithin(16.0, 0.000001));
+    f.checkScalarApprox("log2(-2)", "DOUBLE NOT NULL",
+        "NaN");
+    f.checkScalarApprox("log2(2/3)", "DOUBLE NOT NULL",
```

Comment:
wait what, why is log2(2/3) `-Infinity`?

Reply:
Using desmos, this should equal ~-0.5849625007

Reply:
It seems that there is a little bug in parsing SqlNode. It seems that log10 also has the same problem. I will open a jira to record the problem.

Reply:
I put the test into log10 and had the same problem

Reply:
https://issues.apache.org/jira/browse/CALCITE-6232, Julian seems to think this is not a bug

Reply:
I mean it's definitely a bug but you may want to rewrite your case such that the issue is more clearly stated. I would say stick with a single test case, give the expected results and the actual results. Feel free to tag me on any cases or PRs you file.

I should have found this during my previous LOG work but I didn't. So let me know how I can help.

Reply:
@tanclary I specially added a test for this bug, I believe that users and developers can understand